modules/navier_stokes/src/auxkernels/WallFunctionWallShearStressAux.C

```diff
@@ -0,0 +1,137 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WallFunctionWallShearStressAux.h"
+
+registerMooseObject("NavierStokesApp", WallFunctionWallShearStressAux);
+
+ADReal
+findUStar3(Real mu, Real rho, ADReal u, Real dist)
+{
+  constexpr int MAX_ITERS{50};
+  constexpr Real REL_TOLERANCE{1e-6};
+
+  constexpr Real von_karman{0.4187};
+
+  Real nu = mu / rho;
+
+  ADReal u_star = std::sqrt(nu * u / dist);
+
+  for (int i = 0; i < MAX_ITERS; ++i)
```

Comment:
Add a comment about what you're doing in the loop like "Newton solve for the intersection..."